kotlin/src/main/kotlin/redux/api/Store.kt

```diff
@@ -0,0 +1,159 @@
+     *
+     * @param[subscriber] The subscriber function
+     * @return A subscription
+     */
+    fun subscribe(subscriber: () -> Unit) = subscribe(Subscriber(subscriber))
+
+    /**
+     * Replaces the reducer currently used by the store to calculate the state.
+     *
+     * @see <a href="http://redux.js.org/docs/api/Store.html#replaceReducer">http://redux.js.org/docs/api/Store.html#replaceReducer</a>
+     *
+     * @param[reducer] The reducer
+     */
+    fun replaceReducer(reducer: Reducer<S>)
+
+    /**
+     * An interface that creates a Redux store.
+     *
+     * @see <a href="http://redux.js.org/docs/Glossary.html#store-creator">http://redux.js.org/docs/Glossary.html#store-creator</a>
+     */
+    interface Creator {
+
+        /**
+         *
+         */
+        fun <S : Any> create(reducer: Reducer<S>, initialState: S, enhancer: Enhancer? = null): Store<S>
+    }
+
+    /**
+     * An interface that composes a store creator to return a new, enhanced store creator.
+     *
+     * @see <a href="http://redux.js.org/docs/Glossary.html#store-enhancer">http://redux.js.org/docs/Glossary.html#store-enhancer</a>
+     */
+    interface Enhancer {
+
+        /**
+         *
+         */
+        fun enhance(next: Creator): Creator
+
+        companion object {
+            operator fun invoke(f: (Creator) -> Creator) = object : Enhancer {
+                override fun enhance(next: Creator): Creator = f(next)
+            }
+        }
+    }
+
+    /**
+     * A listener which will be called any time an action is dispatched, and some part of the state tree may potentially
+     * have changed. You may then call [getState()] to read the current state tree inside the listener.
+     *
+     * @see <a href="http://redux.js.org/docs/api/Store.html#subscribe">http://redux.js.org/docs/api/Store.html#subscribe</a>
+     */
+    interface Subscriber {
+
+        /**
+         * Called any time an action is dispatched.
+         */
+        fun onStateChanged()
+
+        companion object {
+
+            /**
+             * Creates a new [Subscriber] instance using the provided function as the [onStateChanged()] implementation.
+             *
+             * @param[f] A higher-order function equivalent to the [onStateChanged()] function
+             * @return A new subscriber instance
+             */
+            operator fun invoke(f: () -> Unit) = object : Subscriber {
+                override fun onStateChanged() = f()
+            }
+
+        }
+
+    }
+
+    /**
+     * A reference to the [Subscriber] to allow for unsubscription.
+     */
+    interface Subscription {
+
+        /**
+         * Unsubscribe the [Subscriber] from the [Store].
+         */
+        fun unsubscribe()
+
+        companion object {
+            operator fun invoke(f: () -> Unit) = object : Subscription {
+                override fun unsubscribe() = f()
+            }
+
+        }
+    }
+
+    companion object {
```

Comment:
`Store` does not provide a `createStore`method because it is considered as logic.